File: api/src/main/java/com/ruby/rubia_server/core/dto/CustomerDTO.java
package com.ruby.rubia_server.core.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CustomerDTO {
    
    private UUID id;
    private UUID companyId;
    private String phone;
    private String name;
    private String whatsappId;
    private String profileUrl;
    private Boolean isBlocked;
    private String sourceSystemName;
    private String sourceSystemId;
    private LocalDateTime importedAt;
    private LocalDate birthDate;
    private LocalDate lastDonationDate;
    private LocalDate nextEligibleDonationDate;
    private String bloodType;
    private Integer height;
    private Double weight;
    private String addressStreet;
    private String addressNumber;
    private String addressComplement;
    private String addressPostalCode;
    private String addressCity;
    private String addressState;
    private String email;
    private String cpf;
    private String rg;
    private String rhFactor;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: api-dotnet/Rubia.Server/DTOs/CustomerDto.cs
namespace Rubia.Server.DTOs;

public class CustomerDto
{
    public Guid Id { get; set; }
    public Guid CompanyId { get; set; }
    public string Phone { get; set; } = string.Empty;
    public string? Name { get; set; }
    public string? WhatsappId { get; set; }
    public string? ProfileUrl { get; set; }
    public bool IsBlocked { get; set; }
    public string? SourceSystemName { get; set; }
    public string? SourceSystemId { get; set; }
    public DateTime? ImportedAt { get; set; }
    public DateOnly? BirthDate { get; set; }
    public DateOnly? LastDonationDate { get; set; }
    public DateOnly? NextEligibleDonationDate { get; set; }
    public string? BloodType { get; set; }
    public int? Height { get; set; }
    public double? Weight { get; set; }
    public string? AddressStreet { get; set; }
    public string? AddressNumber { get; set; }
    public string? AddressComplement { get; set; }
    public string? AddressPostalCode { get; set; }
    public string? AddressCity { get; set; }
    public string? AddressState { get; set; }
    public string? Email { get; set; }
    public string? Cpf { get; set; }
    public string? Rg { get; set; }
    public string? RhFactor { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}